File: apps/mobile/modules/lazynext-native/android/src/main/java/expo/modules/mymodule/MyModule.kt
// MyModule.kt — Lazynext Android native bridge.
// Exposes project info, AI intent processing, and clip CRUD to React Native
// via Expo Modules.  Delegates to the Rust UniFFI bindings for core logic.

package expo.modules.mymodule

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import uniffi.lazynext_mobile.*

class MyModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MyModule")

    AsyncFunction("getProjectInfo") { ->
      getProjectInfo()
    }

    AsyncFunction("processIntent") { prompt: String, requireApproval: Boolean ->
      processIntent(prompt, requireApproval)
    }

    AsyncFunction("addClip") { trackIndex: Long, clipType: String, name: String, start: Long, end: Long ->
      addClip(trackIndex.toUInt(), clipType, name, start.toUInt(), end.toUInt())
    }

    AsyncFunction("moveClip") { clipId: String, newStart: Long ->
      moveClip(clipId, newStart.toUInt())
    }
  }
}

File: apps/mobile/modules/lazynext-native/ios/MyModule.swift
import ExpoModulesCore

public class MyModule: Module {
  public func definition() -> ModuleDefinition {
    Name("MyModule")

    AsyncFunction("getProjectInfo") { () -> String in
      return try getProjectInfo()
    }

    AsyncFunction("processIntent") { (prompt: String, requireApproval: Bool) -> String in
      return try processIntent(prompt: prompt, requireApproval: requireApproval)
    }

    AsyncFunction("addClip") { (trackIndex: UInt32, clipType: String, name: String, start: UInt32, end: UInt32) -> String in
      return try addClip(trackIndex: trackIndex, clipType: clipType, name: name, start: start, end: end)
    }

    AsyncFunction("moveClip") { (clipId: String, newStart: UInt32) -> String in
      return try moveClip(clipId: clipId, newStart: newStart)
    }
  }
}
